static/app/views/settings/projectTags.spec.jsx

```diff
@@ -38,42 +48,37 @@ describe('ProjectTags', function () {
       body: [],
     });
 
-    const wrapper = mountWithTheme(
-      <ProjectTags params={{orgId: org.slug, projectId: project.slug}} />
-    );
+    render(<ProjectTags params={{orgId: org.slug, projectId: project.slug}} />);
 
-    expect(wrapper.find('EmptyMessage')).toHaveLength(1);
+    expect(screen.getByText(/There are no tags/)).toBeInTheDocument();
   });
 
   it('disables delete button for users without access', function () {
-    const context = {
-      organization: TestStubs.Organization({access: []}),
-    };
-
-    const wrapper = mountWithTheme(
-      <ProjectTags params={{orgId: org.slug, projectId: project.slug}} />,
-      TestStubs.routerContext([context])
-    );
+    render(<ProjectTags params={{orgId: org.slug, projectId: project.slug}} />, {
+      context: TestStubs.routerContext([
+        {organization: TestStubs.Organization({access: []})},
+      ]),
+    });
 
-    expect(wrapper.find('Button[disabled=false]')).toHaveLength(0);
+    expect(screen.getAllByLabelText('Remove tag').at(0)).toBeDisabled();
   });
 
   it('deletes tag', async function () {
-    const wrapper = mountWithTheme(
-      <ProjectTags params={{orgId: org.slug, projectId: project.slug}} />
-    );
+    render(<ProjectTags params={{orgId: org.slug, projectId: project.slug}} />);
+    renderGlobalModal();
 
-    const tags = wrapper.state('tags').length;
-
-    wrapper.find('button[data-test-id="delete"]').first().simulate('click');
+    expect(screen.getAllByTestId('tag-row')).toHaveLength(tags.length);
+    userEvent.click(screen.getAllByRole('button', {name: 'Remove tag'}).at(0));
 
     // Press confirm in modal
-    const modal = await mountGlobalModal();
-    modal.find('Button[priority="primary"]').simulate('click');
-
-    await tick(); // Wait for the handleDelete function
-    wrapper.update();
+    userEvent.click(screen.getByRole('button', {name: 'Confirm'}));
 
-    expect(wrapper.state('tags')).toHaveLength(tags - 1);
+    await waitFor(() =>
+      expect(screen.getAllByTestId('tag-row')).toHaveLength(tags.length - 1)
+    );
```

Comment:
Complete aside: how do you feel about awaiting the user event interactions? I kinda feel like `waitFor` is less magic then remembering awaiting the interaction, but if we consistently await all interactions it would make tests all read the same / simpler.


```js
   await userEvent.click(screen.getByRole('button', {name: 'Confirm'}));
   expect(screen.getAllByTestId('tag-row')).toHaveLength(tags.length - 1)
```

Reply:
I agree that the code would be visually simpler, but the line of reasoning compared to what would happen, doesn't match.

Usually, we users click on something, a button for instance that triggers something, and then `wait` for the UI to give feedback as a response to this action. In the above code would be fewer rendered tags in the dom.

Also`userEvent.click` doesn't return a promise...was that just an idea?

Reply:
Newer version of userEvents supports await on interaction